### PR TITLE
Werad penalty

### PR DESCRIPTION
**Pull request by weradsaoud**

### CODE REVIEW

**Review by BeSaRa — CHANGES_REQUESTED** [review]
reolve it and resubmit

File: src/abstracts/admin-component.ts [hunk 1]
```diff
@@ -109,17 +115,27 @@ export abstract class AdminComponent<
...
+              // )
+            )
           );
         })
       );
   }
 
   ngOnInit(): void {
+    this.data$.subscribe((data) => {
+      console.log('data: ', data);
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/abstracts/admin-component.ts:133]: done

File: src/abstracts/admin-dialog-component.ts [hunk 2]
```diff
@@ -79,6 +79,8 @@ export abstract class AdminDialogComponent<
       )
       .pipe(
         exhaustMap((model) => {
+          console.log('to save model: ', model);
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/abstracts/admin-dialog-component.ts:82]: done

File: src/app/app.module.ts [hunk 3]
```diff
@@ -35,6 +37,8 @@ import { MenuItemService } from '@services/menu-item.service';
 import { NavbarComponent } from '@standalone/components/navbar/navbar.component';
 import { MatPaginatorIntl } from '@angular/material/paginator';
 import { PaginatorLocal } from '@constants/paginator-local';
+import { MAT_LEGACY_SNACK_BAR_DEFAULT_OPTIONS } from '@angular/material/legacy-snack-bar';
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/app/app.module.ts:40]: done

File: src/modules/administration/components/penalty/penalty.component.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  styleUrls: ['./penalty.component.scss'],
+})
+export class PenaltyComponent extends AdminComponent<
+  PenaltyPopupComponent,
+  Penalty,
+  PenaltyService
+> {
+  filter_Value!: string;
+  service = inject(PenaltyService);
+  displayedColumns: string[] = [
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/administration/components/penalty/penalty.component.ts:24]: done

File: src/modules/administration/components/penalty/penalty.component.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        this.delete$.next(item);
+      },
+    },
+  ];
+  // here we have a new implementation for displayed/filter Columns for the table
+  columnsWrapper: ColumnsWrapper<Penalty> = new ColumnsWrapper(
+    new NoneFilterColumn('select'),
+    new TextFilterColumn('arName'),
+    new TextFilterColumn('enName'),
+    new NoneFilterColumn('status'),
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/administration/components/penalty/penalty.component.ts:74]: done

File: src/modules/administration/popups/penalty-popup/penalty-popup.component.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        <app-switch
+          formControlName="status"
+          [trueValue]="1"
+          [falseValue]="0"
+          [label]="lang.map.change_status"
+          [matTooltip]="statusTooltipText"></app-switch>
+        <!-- <div class="w-100 h-100 flex text-center justify-start">
+          <mat-slide-toggle formControlName="status" labelPosition="before" [name]="lang.map.change_status"
+            [checked]="model ? model.isActive() : false">{{ lang.map.change_status }}</mat-slide-toggle>
+        </div> -->
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/administration/popups/penalty-popup/penalty-popup.component.html:26]: done

File: src/services/lookup.service.ts [hunk 7]
```diff
@@ -9,6 +9,8 @@ export class LookupService {
   lookups: LookupMapContract = {} as LookupMapContract;
 
   setLookups(lookups: LookupMapContract) {
+    console.log('lookups: ', lookups);
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/services/lookup.service.ts:12]: done

File: src/standalone/components/context-menu/context-menu.component.ts [hunk 8]
```diff
@@ -67,7 +67,7 @@ export class ContextMenuComponent {
     const parent = target.parentElement;
     const isTR = () => parent?.nodeName === 'TR';
 
-    this.trigger.menuOpened.pipe(take(1)).subscribe(() => {
+    this.trigger.menuOpened.pipe(take(2)).subscribe(() => {
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/standalone/components/context-menu/context-menu.component.ts:70]: done

File: src/modules/administration/popups/penalty-popup/penalty-popup.component.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  styleUrls: ['./penalty-popup.component.scss'],
+})
+export class PenaltyPopupComponent extends AdminDialogComponent<Penalty> {
+  lookupService = inject(LookupService);
+
+  form!: UntypedFormGroup;
+  data: CrudDialogDataContract<Penalty> = inject(MAT_DIALOG_DATA);
+
+  penaltyTypes: { name: string; lookupKey: number }[] =
+    this.getTodisplayPenaltyTypes();
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/administration/popups/penalty-popup/penalty-popup.component.ts]: done

File: src/modules/administration/popups/penalty-popup/penalty-popup.component.html [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  <div class="dialog-header bg-primary">
+    <h1>{{ lang.map.menu_penalty }}</h1>
+    <app-icon-button tabindex="-1" mat-dialog-close="" class="close-btn" icon="CLOSE"></app-icon-button>
+  </div>
+  <div class="dialog-content">
+    <div class="grid grid-cols-2 gap-4">
+      <ng-container [formGroup]="form">
+        <app-input formControlName="arName" [label]="lang.map.ar_name"></app-input>
+        <app-input formControlName="enName" [label]="lang.map.en_name"></app-input>
+        <app-select-input
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/administration/popups/penalty-popup/penalty-popup.component.html:11]: I used the most recent implementation of select-input

**Pull request closed without merging**